api/v1beta1/aerospikecluster_validating_webhook.go

```diff
@@ -920,16 +1162,36 @@ func validateRequiredFileStorage(aslog logr.Logger, configSpec AerospikeConfigSp
 	return nil
 }
 
-func validateConfigMapVolumes(configSpec AerospikeConfigSpec, storage *AerospikeStorageSpec, validationPolicy *ValidationPolicySpec, version string) error {
-	_, err := storage.GetConfigMaps()
-	return err
+func validateRequiredFileStorageForFeatureConf(
+	aslog logr.Logger, configSpec AerospikeConfigSpec,
+	storage *AerospikeStorageSpec,
+) error {
+	// TODO Add validation for feature key file.
+	featureKeyFilePaths := getFeatureKeyFilePaths(configSpec)
+	tlsPaths := getTLSFilePaths(configSpec)
+
+	var allPaths []string
+	allPaths = append(allPaths, featureKeyFilePaths...)
+	allPaths = append(allPaths, tlsPaths...)
+
+	for _, path := range allPaths {
+		if !storage.IsVolumePresentForAerospikePath(filepath.Dir(path)) {
```

Comment:
This will break validation for use cases when credentials are feed not through secret.

Reply:
@skolosov-snap This only checks if the path has an associated volume attached. The volume could be any one of a secret, a config map, a shared persistent volume, or a shared empty dir.